src/neuroconv/datainterfaces/ophys/miniscope/miniscopedatainterface.py

```diff
@@ -0,0 +1,63 @@
+import os
+from typing import Optional
+from glob import glob
+
+from natsort import natsorted
+from pynwb.file import NWBFile
+from pynwb.image import ImageSeries
+
+from ....basedatainterface import BaseDataInterface
+from ....tools.nwb_helpers import make_or_load_nwbfile
+
+
+class MiniscopeImagingInterface(BaseDataInterface):
+    def __init__(self, data_dir: str):
+        super().__init__(data_dir=data_dir)
+
+    def run_conversion(
+        self,
+        nwbfile_path: Optional[str] = None,
+        nwbfile: Optional[NWBFile] = None,
+        metadata: Optional[dict] = None,
+        overwrite: bool = False,
+        verbose=False,
+    ):
+        from ndx_miniscope import read_settings, read_notes, read_miniscope_timestamps, get_starting_frames
+
+        data_dir = self.source_data["data_dir"]
+        miniscope = read_settings()
+        annotations = read_notes(data_dir)
+
+        ms_files = natsorted(glob(os.path.join(data_dir, "msCam*.avi")))
+        behav_files = natsorted(glob(os.path.join(data_dir, "behavCam*.avi")))
+
+        with make_or_load_nwbfile(
+            nwbfile_path=nwbfile_path,
+            nwbfile=nwbfile,
+            metadata=metadata,
+            overwrite=overwrite,
+        ) as nwb:
+
+            nwb.add_device(miniscope)
+            if annotations is not None:
+                nwb.add_acquisition(annotations)
+
+            nwb.add_acquisition(
+                ImageSeries(
+                    name="OnePhotonSeries",
+                    format="external",
+                    external_file=[os.path.split(x)[1] for x in ms_files],
+                    timestamps=read_miniscope_timestamps(data_dir),
+                    starting_frame=get_starting_frames(ms_files),
+                )
+            )
+
+            nwb.add_acquisition(
+                ImageSeries(
+                    name="behaviorCam",
```

Comment:
Gonna need some example data together with some example stub files to really understand this, but the behavior video stream is attached to the regular microscope camera? Are they synchronized through the same system?

Reply:
http://miniscope.org/index.php/Data_Acquisition_Software#Data_Structure

Reply:
That is pretty nice then actually, first time I've seen a format with auto-synchronizing behavior stream!